# Weghted Ensemble method of Spatial Transcriptomics (WEST) 

## 1. Introduction 
WEST is a method that utilizes ensemble techniques to improve the performance and stability of spatial transcriptomics data analysis. 
It represents a significant advance in clustering spatial transcriptomics data, offering improved accuracy and flexibility compared to existing methods, 
making it a valuable tool for spatial transcriptomics data analysis.

## 2. Environment 
Python=3.12.4 
### Required packages: 
The detailed requirements can be found [here](https://github.com/JiazhangCai/WEST/blob/main/requirements.txt).

## 3.Tutorial 
The [tutorial](https://github.com/JiazhangCai/WEST/blob/main/tutorial/WEST.ipynb) provides a detailed tutorial on how to implement
WEST on an actual sample, including the description for every parameter. The input of WEST is the embeddings generated by other deep learning-based 
methods. Since deep learning-based methods usually require an extremely complicated environment, we only provide the required packages to run the WEST pipeline. To run the tutorial, we provide the embeddings from SpaGCN and SpaceFlow in [embedding](https://github.com/JiazhangCai/WEST/tree/main/embedding). We also provide the tutorial for generating the embedding, but the user needs to construct the environment according to the requirements of [SpaGCN](https://github.com/jianhuupenn/SpaGCN) and [SpaceFlow](https://github.com/hongleir/SpaceFlow).

## 4. Data used in the paper 

### 4.1 Simulation data

The code for simulation data is provided. [Square](https://github.com/JiazhangCai/WEST/blob/main/simulation/simulation_square.R) is the code 
for generating simulation data with the squared spatial pattern, and [circle](https://github.com/JiazhangCai/WEST/blob/main/simulation/simulation_circle.R) 
is the code for generating simulation data with the circled spatial pattern.

### 4.2 Real data

The DLPFC data used in the paper can be found [here](http://research.libd.org/spatialLIBD/).     
The mouse embryo data used in the paper can be found [here](https://crukci.shinyapps.io/SpatialMouseAtlas/).  
The 10X Genomics Visium fluorescence image data of the mouse brain used in the paper can be found 
[here](https://support.10xgenomics.com/spatial-gene-expression/datasets).
The 10X Genomics Visium H&E image data of the mouse brain used in the paper can be found 
[here](https://support.10xgenomics.com/spatial-gene-expression/datasets).
The human pancreatic cancer data used in the paper can be found [here](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE111672).
 

