# Weghted Ensemble method of Spatial Transcriptomics (WEST) 

## 1. Introduction 
WEST is a method that utilizes ensemble techniques to improve the performance and stability of spatial transcriptomics data analysis. 
It represents a significant advance in clustering spatial transcriptomics data, offering improved accuracy and flexibility compared to existing methods, 
making it a valuable tool for spatial transcriptomics data analysis.

## 2. Environment 
Python=3.8 
### Required packages: 
The detailed requirements can be found [here](https://github.com/JiazhangCai/WEST/blob/main/requirements.txt).

## 3.Tutorial 
The [tutorial](https://github.com/JiazhangCai/WEST/blob/main/tutorial/WEST.ipynb) provides a detailed tutorial on how to implement
WEST on an actual sample, including the description for every parameter. The input of WEST is the embeddings generated by other deep learning-based 
methods. The tutorials on implementing the six benchmarking methods: [Leiden](https://github.com/JiazhangCai/WEST/blob/main/tutorial/Leiden.ipynb),
[SpaGCN](https://github.com/JiazhangCai/WEST/blob/main/tutorial/SpaGCN.ipynb), 
[SpaceFlow](https://github.com/JiazhangCai/WEST/blob/main/tutorial/SpaceFlow.ipynb), 
[SEDR](https://github.com/JiazhangCai/WEST/blob/main/tutorial/SEDR.ipynb), 
[STAGATE](https://github.com/JiazhangCai/WEST/blob/main/tutorial/STAGATE.ipynb), 
and [DeepST](https://github.com/JiazhangCai/WEST/blob/main/tutorial/DeepST.ipynb)
are also provided.
[Support resouce](https://github.com/JiazhangCai/WEST/tree/main/support_resources) provides some required code for the benchmarking methods.

## 4. Data used in the paper 

### 4.1 Simulation data

The code for simulation data is provided. [Square](https://github.com/JiazhangCai/WEST/blob/main/simulation/simulation_square.R) is the code 
for generating simulation data with the squared spatial pattern, and [circle](https://github.com/JiazhangCai/WEST/blob/main/simulation/simulation_circle.R) 
is the code for generating simulation data with the circled spatial pattern.

### 4.2 Read data

The DLPFC data used in the paper can be found [here](http://research.libd.org/spatialLIBD/).     
The mouse embryo data used in the paper can be found [here](https://crukci.shinyapps.io/SpatialMouseAtlas/).  
The 10X Genomics Visium fluorescence image data of the mouse brain used in the paper can be found 
[here](https://support.10xgenomics.com/spatial-gene-expression/datasets).
The 10X Genomics Visium H&E image data of the mouse brain used in the paper can be found 
[here](https://support.10xgenomics.com/spatial-gene-expression/datasets).
The human pancreatic cancer data used in the paper can be found [here](https://www.ncbi.nlm.nih.gov/geo/query/acc.cgi?acc=GSE111672).
 

